### PR TITLE
Storing and loading neural networks from file

### LINKED ISSUE #15: Saving and loading neural networks
Application should be able to save neural networks on disc to load it later.

### PR DESCRIPTION
**Pull request by ErykKrupa**

### CODE REVIEW

File: src/model/car.py [hunk 1]
```diff
@@ -79,11 +79,8 @@ def _update_speed(self, acceleration: float, delta_time: float) -> None:
 
     def tick(self, track: Track, delta_time: float) -> None:
         distances = self._sense_surroundings(track)
-        turning_rate, acceleration = self.neural_network.compute(np.array(distances))
+        turning_rate, acceleration = self.neural_network.predict(np.array(distances).reshape(1, len(distances)))[0]
```
Comment (jakubdabek) [inline src/model/car.py]:
```suggestion
        turning_rate, acceleration = self.neural_network.predict(np.array(distances).reshape(1, -1))[0]
```

File: src/model/neural_network/neural_network_store.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+             directory_name: str = _DEFAULT_DIRECTORY,
+             amount_to_load: int = -1) -> List[NeuralNetwork]:
+        """
+        Loads neural networks from binary file with .nn extension.
+
+        Args:
+            file_name (str): Name of file to load neural networks from
+            directory_name (str): Name of directory where file to load is.
+            amount_to_load (int): Maximal amount of neural networks to load,
+                if there is no enough neural networks in file,
```
Comment (jakubdabek) [inline src/model/neural_network/neural_network_store.py]:
```suggestion
                if there is not enough neural networks in file,
```

File: src/model/neural_network/neural_network_store.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                everything will be loaded without any exception,
+                same for negative value.
+        Raises:
+            FileNotFoundError: If file or directory name doesn't exist.
```
Comment (jakubdabek) [inline src/model/neural_network/neural_network_store.py]: Names always exist as strings, it's files and directories themselves that don't always exist.

File: src/model/neural_network/neural_network_store.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+              directory_name: str = _DEFAULT_DIRECTORY) -> None:
+        """
+        Stores neural networks in binary file with .nn extension.
+
+        Args:
+            neural_networks (List[NeuralNetwork]): Neural networks to store in file.
+            file_name (str): Name of file to store neural networks in,
+                will be created if doesn't exist.
+            directory_name (str): Name of directory where file is,
+                will be created if doesn't exist.
```
Comment (jakubdabek) [inline src/model/neural_network/neural_network_store.py]:
```suggestion
            file_name (str): Name of file to store neural networks in,
                will be created if it doesn't exist.
            directory_name (str): Name of directory where the file is,
                will be created if it doesn't exist.
```

**Review by jakubdabek — CHANGES_REQUESTED** [review]
Evidently `pre-commit` has not been run on this commit - mypy can't recognize modules without `__init__.py` anymore and changes aren't formatted in the standard style. Please check every commit you make.

File: src/model/neural_network/neural_network_store.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        if not exists(directory_name):
+            makedirs(directory_name)
+        with open(f'{directory_name}/{file_name}.nn', 'wb') as store:
+            for neural_network in neural_networks:
+                pickle.dump(neural_network, store, pickle.HIGHEST_PROTOCOL)
+
+    @staticmethod
+    def load(file_name: str,
+             directory_name: str = _DEFAULT_DIRECTORY,
+             amount_to_load: int = -1) -> List[NeuralNetwork]:
```
Comment (KamilKrol5) [inline src/model/neural_network/neural_network_store.py:38]:
```suggestion
    def load(file_name: str,
             directory_name: str = _DEFAULT_DIRECTORY,
             amount_to_load: int = -1
    ) -> List[NeuralNetwork]:
```

**Review by KamilKrol5 — APPROVED** [review]

**Review by jakubdabek — APPROVED** [review]

**Pull request merged**